Parser/Number.elm

```diff
@@ -34,9 +35,17 @@ integer : Parser Int
 integer =
     map (\sig nat -> sig * nat) sign `and` natural
 
+{-| The fromOk function extracts the element out of a Ok. -}
+fromOk : Result e a -> a
+fromOk r = case r of
+    Ok n -> n
+
 {-| Parse a float with optional sign -}
 float : Parser Float
 float =
-    let convertToFloat sig int dig = toFloat sig * (toFloat int + 0.1 * List.foldr (\b a -> a / 10 + b) 0 (List.map toFloat dig))
+    let toFloatString (i,ds) =
+          toString i ++ List.foldl (\d s -> s ++ toString d) "." ds
+        convertToFloat sig int digs =
+          toFloat sig * (fromOk << String.toFloat << toFloatString) (int,digs)
```

Comment:
@Dandandan I followed your suggestion and now I'm converting to string and then using `String.toFloat`. Since we have a `Result e a`, we need to extract out the `a` from Ok. OTOH, there's the `e` case. I believe the parser will give us a valid float and at this point we shouldn't worry about that case. I think the reasonable approach would be to fail somehow instead of providing a default value if we see this exception (in theory shouldn't happen). What do you think about the use of unsafe `fromOk` in this particular case? Is there anything else we could improve? :)